### PR TITLE
fix: Adding handling for columns with multiple words in name for filtering data in filtering library

### LINKED ISSUE #7599: [Bug] Google Sheets where clause does not support space separated column name (aka multiple words)
If the column name comprises of multiple words, the insertion of data into H2 database fails leading to errors.

### PR DESCRIPTION
**Pull request by trishaanand**
Fixes #7599

### CODE REVIEW

File: app/server/appsmith-interfaces/src/test/java/com/appsmith/external/services/FilterDataServiceTest.java [hunk 1]
```diff
@@ -274,6 +274,53 @@ public void testFilterNotInConditionForNumbers() {
...
+            ArrayNode items = (ArrayNode) objectMapper.readTree(data);
+
+            List<Condition> conditionList = new ArrayList<>();
+
+            Condition condition = new Condition("orderAmount", "LT", "15");
+            conditionList.add(condition);
+
+            ArrayNode filteredData = filterDataService.filterData(items, conditionList);
+
+            assertEquals(filteredData.size(), 2);
```
Comment (nidhi-nair) [inline app/server/appsmith-interfaces/src/test/java/com/appsmith/external/services/FilterDataServiceTest.java:321]: Minor could you also add a check for the upper case conversion here?

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/services/FilterDataService.java [hunk 2]
```diff
@@ -298,6 +300,7 @@ private void executeDbQuery(String query) {
 
         Connection conn = checkAndGetConnection();
 
+        log.debug("{}", query);
```
Comment (nidhi-nair) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/services/FilterDataService.java:303]: Is this required?

Comment (trishaanand) [reply app/server/appsmith-interfaces/src/main/java/com/appsmith/external/services/FilterDataService.java:303]: Issh. Sorry. Removing this.

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/services/FilterDataService.java [hunk 3]
```diff
@@ -180,7 +180,7 @@ private String generateWhereClause(List<Condition> conditions) {
                         operator.toString() + " is not supported currently for filtering.");
             }
 
-            sb.append(path);
+            sb.append("\"" + path + "\"");
```
Comment (nidhi-nair) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/services/FilterDataService.java:183]: QQ: How would this work for anything with quotes in them?

Comment (trishaanand) [reply app/server/appsmith-interfaces/src/main/java/com/appsmith/external/services/FilterDataService.java:183]: No good solution that I can come up with right now. Throwing an exception if the column name contains quotes. Handling values via Prepared Statement for which there is a raised issue.

**Pull request merged**